Classes/KIFUITestActor.m

```diff
@@ -468,24 +468,25 @@ - (void)enterTextIntoCurrentFirstResponder:(NSString *)text fallbackView:(UIView
     [text enumerateSubstringsInRange:NSMakeRange(0, text.length)
                              options:NSStringEnumerationByComposedCharacterSequences
                           usingBlock: ^(NSString *characterString,NSRange substringRange,NSRange enclosingRange,BOOL * stop)
-    {
+     {
         if (![KIFTypist enterCharacter:characterString]) {
+            NSLog(@"KIF: Unable to find keyboard key for %@. Will attempt to insert manually.", characterString);
+
             // Attempt to cheat if we couldn't find the character
-            UIView * fallback = fallbackView;
-            if (!fallback) {
-                UIResponder *firstResponder = [[[UIApplication sharedApplication] keyWindow] firstResponder];
+            NSArray *fallbackViews = fallbackView ? @[fallbackView] : [[UIApplication sharedApplication] firstResponders];
+
+            for (id fallback in fallbackViews) {
+                if ([fallback isKindOfClass:[UITextField class]] || [fallback isKindOfClass:[UITextView class]] || [fallback isKindOfClass:[UISearchBar class]]) {
+                    if (fallbackViews.count > 1) {
```

Comment:
I was thinking of gathering the set of first responders that match the class type check and then output the log message only if there is more than one match. Just concerned that this might be too noisy if it is an expected common scenario.

Reply:
Fixed!